### PR TITLE
solution

### PR DESCRIPTION
**Pull request by 1stIndominus**
[DEMO LINK](https://1stIndominus.github.io/react_todo-app-add-and-delete/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,222 +1,183 @@
...
+    });
+
+    setShowAll(!showAll);
+  };
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    getTodos(user?.id)
+      .then(res => setTodos(res))
+      .catch(() => isError);
```
Comment (Anastasiia-Svintsova) [inline src/App.tsx:96]: Do you understand what this catch will do? Maybe you need to use `setIsError` here instead?

File: src/components/ErrorType/ErrorType.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
```
Comment (Anastasiia-Svintsova) [inline src/components/ErrorType/ErrorType.tsx:1]: don't disable eslint

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  };
+
+  const optionSelected = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    setFilter(event.currentTarget.innerHTML.toLocaleLowerCase());
+  };
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="todosCounter">
+        {`${todos.length - completedTodos.length} items left`}
```
Comment (Anastasiia-Svintsova) [inline src/components/Footer/Footer.tsx:35]: Let's move `todos.length - completedTodos.length` to a variable

File: src/components/Footer/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          data-cy="FilterLinkAll"
+          href="#/"
+          className={classNames('filter__link',
+            { 'filter__link selected': filter === 'all' })}
```
Comment (Anastasiia-Svintsova) [inline src/components/Footer/Footer.tsx:43]: You have the enum for `all` value

File: src/components/Footer/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+          onClick={optionSelected}
+        >
+          All
+        </a>
+
+        <a
+          data-cy="FilterLinkActive"
+          href="#/active"
+          className={classNames('filter__link', {
+            'filter__link selected': filter === 'active',
```
Comment (Anastasiia-Svintsova) [inline src/components/Footer/Footer.tsx:53]: And for `active`. Use the enum

File: src/components/Footer/Footer.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+          })}
+          onClick={optionSelected}
+        >
+          Active
+        </a>
+        <a
+          data-cy="FilterLinkCompleted"
+          href="#/completed"
+          className={classNames('filter__link', {
+            'filter__link selected': filter === 'completed',
```
Comment (Anastasiia-Svintsova) [inline src/components/Footer/Footer.tsx:63]: Same for `completed`

File: src/components/TodoList/TodoList.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (Anastasiia-Svintsova) [inline src/components/TodoList/TodoList.tsx:1]: Same about eslint. Check it everywhere

**Review by Anastasiia-Svintsova — CHANGES_REQUESTED** [review]
- The issue with the loader still exists. You should show the new todo with the loader, not just the loader. Check the [working example](https://mate-academy.github.io/react_todo-app-with-api/). If you have any questions feel free to ask them in the fe_chat.

https://user-images.githubusercontent.com/99403198/206853989-e035c1fc-c62d-439e-be3a-38e120f394e5.mov

File: src/App.tsx [hunk 8]
```diff
@@ -1,222 +1,183 @@
...
+    } catch (error) {
+      handleError('Can not delete a todo');
+    }
+  }, []);
+
+  const changeTodo = useCallback(async (todoId: number, object: TodoTitle) => {
+    try {
+      const updatedTodo: Todo = await updateTodo(todoId, object);
+
+      setTodos(prevState => (prevState.map((a) => (a.id === todoId
```
[the remark this answers is not in the document]
Comment (loralevitska) [reply src/App.tsx:86]: not fixed yet

File: src/App.tsx [hunk 9]
```diff
@@ -1,222 +1,208 @@
...
+    }, 1000);
+
+    setShowAll(!showAll);
+  };
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    getTodos(user?.id)
+      .then(res => setTodos(res))
+      .catch(() => setIsError);
```
Comment (loralevitska) [inline src/App.tsx:116]: What did you want to express in the code with this entry? Can you explain please?
```suggestion
      .catch((error) => setIsError(error?.message));
```

Comment (loralevitska) [reply src/App.tsx:116]: I recommend that you repeat [`try {...} catch {...}`](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Statements/try...catch)

**Review by loralevitska — CHANGES_REQUESTED** [review]
GJ) A little more and it will be completed

File: src/App.tsx [hunk 10]
```diff
@@ -1,222 +1,208 @@
...
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState<string>(SelectedType.ALL);
+  const [showAll, setShowAll] = useState(true);
+  const [isDeleting, setIsDeleting] = useState<number[]>([]);
+  const [isChangeAllTodos, setIsChangeAllTodos] = useState<number[]>([]);
+  const [isError, setIsError] = useState('');
```
Comment (polosanya) [inline src/App.tsx:29]: Prefix `is` is usually used in names for boolean values. It can be confusing for other developers to put it for other values

**Review by polosanya — CHANGES_REQUESTED** [review]
Good job! I'm approving it, but please pay attention to my comment

**Pull request closed without merging**